components/suggest/src/db.rs

```diff
@@ -294,6 +295,33 @@ impl<'a> SuggestDao<'a> {
                             Ok(None)
                         }
                     },
+                    SuggestionProvider::Mdn => {
+                        let full_suffix = row.get::<_, String>("keyword_suffix")?;
+                        self.conn.query_row_and_then(
+                            r#"
+                             SELECT
+                                 description
+                             FROM
+                                 mdn_custom_details
+                             WHERE
+                                 suggestion_id = :suggestion_id
+                            "#,
+                            named_params! {
+                                ":suggestion_id": suggestion_id
+                            },
+                            |row| {
+                                if full_suffix.starts_with(keyword_suffix) {
```

Comment:
I wonder if we can do this check before we do the `query_row_and_then`? (I think we could do the same for AMO, but that can be a different PR; no need to do that here 😊) No need to query `mdn_custom_details` if the suffixes don't match.

Reply:
Yup, nice catch! Let's do that.